float32[] critic_grad_msg
